### PR TITLE
 Alethe: Added Proof Postprocessor to alethe_proof_processor

### PR DESCRIPTION
**Pull request by Lachnitt**
Added proof postprocessor class to alethe_proof_processor header file.

Should be merged after PR #7200.

Co-authored-by: Haniel Barbosa hanielbbarbosa@gmail.com

### CODE REVIEW

File: src/proof/alethe/alethe_post_processor.h [hunk 1]
```diff
@@ -99,6 +99,69 @@ class AletheProofPostprocessCallback : public ProofNodeUpdaterCallback
...
+  /** post-process */
+  void process(std::shared_ptr<ProofNode> pf);
+
+ private:
+  /** The proof node manager */
+  ProofNodeManager* d_pnm;
+  /** The post process callback */
+  AletheProofPostprocessCallback d_cb;
+  /** The updater, which is responsible for translating proof rules */
+  ProofNodeUpdater d_updater;
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.h:157]: This does not need to be an attribute of the class. It is only gonna be used inside `process`, so it should be a local variable there.

File: src/proof/alethe/alethe_post_processor.h [hunk 2]
```diff
@@ -99,6 +99,69 @@ class AletheProofPostprocessCallback : public ProofNodeUpdaterCallback
...
+  /** The final post process callback */
+  AletheProofPostprocessFinalCallback d_fcb;
+  /** The updater, which is responsible for adding additional steps to the end
+   * of the proof */
+  ProofNodeUpdater d_finalize;
```
Comment (HanielB) [inline src/proof/alethe/alethe_post_processor.h:162]: Ditto

**Review by HanielB — CHANGES_REQUESTED** [review]

File: src/proof/alethe/alethe_post_processor.h (hunk 1 above)
Comment (Lachnitt) [reply src/proof/alethe/alethe_post_processor.h:157]: Done

File: src/proof/alethe/alethe_post_processor.h (hunk 2 above)
Comment (Lachnitt) [reply src/proof/alethe/alethe_post_processor.h:162]: Done

**Review by HanielB — APPROVED** [review]

**Pull request merged**